src/modules/UI/Wallets/action.js

```diff
@@ -74,54 +73,57 @@ export const upsertWallet = (wallet: AbcCurrencyWallet) => (dispatch: any, getSt
+// adds to core and enables in core
+export const addCustomToken = (walletId: string, tokenObj: any) => (dispatch: any, getState: any) => {
+  const state = getState()
+  const wallet = CORE_SELECTORS.getWallet(state, walletId)
+  WALLET_API.addCoreCustomToken(wallet, tokenObj)
+  .then(() => {
+    return
+  })
+  .catch((e) => console.log(e))
+}
+
+export const setEnabledTokens = (walletId: string, enabledTokens: Array<string>, disableTokens: Array<string>) => (dispatch: any, getState: any) => {
+  // tell Redux that we are updating the enabledTokens list
   dispatch(setTokensStart())
+  // get a snapshot of the state
   const state = getState()
+  // get a copy of the relevant core wallet
   const wallet = CORE_SELECTORS.getWallet(state, walletId)
-  WALLET_API.setSyncedTokens(wallet, enabledTokens)
-  .then((tokens) => {
+  // now actually tell the wallet to enable the token(s) in the core and save to file
+  WALLET_API.setEnabledTokens(wallet, enabledTokens, disableTokens)
+  .then(() => {
     dispatch(setTokensSuccess())
     dispatch(refreshWallet(walletId))
-    return tokens
+    wallet.getEnabledTokens()
+    .then((enabledTokens) => {
+      console.log(enabledTokens)
+      return
+    })
+    .catch((e) => console.log(e))
   })
+  .catch((e) => console.log(e))
 }
 
 export const getEnabledTokens = (walletId: string) => (dispatch: any, getState: any) => {
   const state = getState()
   const wallet = CORE_SELECTORS.getWallet(state, walletId)
-  WALLET_API.getSyncedTokens(wallet) // get list of enabled / disbaled tokens on the user's side (not core)
+  WALLET_API.getEnabledTokens(wallet) // get list of enabled / disbaled tokens on the user's side (not core)
   .then((tokens) => {
-    wallet.tokensEnabled = tokens // set the tokensEnabled property on the GuiWallet to the returned tokens object for easy access
-    let guiEnabledTokens = [] // initialize array that will be used to enable tokens in the core
-    for (let prop in tokens) {
-      if (tokens[prop].enabled) {
-        guiEnabledTokens.push(prop)
-        let tokenObj = tokens[prop]
-        tokenObj.multiplier = tokens[prop].denominations[0].multiplier // this needs to be improved upon
-        WALLET_API.addCoreCustomToken(wallet, tokenObj)
-        dispatch(ADD_TOKEN_ACTIONS.setTokenSettings(tokenObj))
-      }
-    }
-    WALLET_API.enableTokens(wallet, guiEnabledTokens) // take GUI enabled tokens and enable them in the core
+    let modifiedWallet = wallet
+    modifiedWallet.enabledTokens = tokens
+    WALLET_API.enableTokens(modifiedWallet, tokens) // take GUI enabled tokens and enable them in the core
     .then(() => {
-      dispatch(upsertWallet(wallet)) // now update the wallet in Redux so that the tokensEnabled property can be used by GUI
+      dispatch(upsertWallet(modifiedWallet)) // now update the wallet in Redux so that the tokensEnabled property can be used by GUI
+      wallet.getEnabledTokens()
```

Comment:
is this being used for anything? ive never used the return value of a dispatched action before.